tasks/Task.cpp

```diff
@@ -13,37 +18,666 @@ Task::~Task()
+        "StreamAutoNegotiatePacketSize",
+        true);
+
+    LOG_INFO_S << "Setting StreamPacketResendEnable." << endl;
+    Arena::SetNodeValue<bool>(m_device->GetTLStreamNodeMap(),
+        "StreamPacketResendEnable",
+        true);
+
+    LOG_INFO_S << "Setting PixelFormat." << endl;
+    Arena::SetNodeValue<GenICam::gcstring>(m_device->GetNodeMap(),
+        "PixelFormat",
+        convertFrameModeToPixelFormat(_image_config.get().format,
+            _image_config.get().depth)
+            .c_str());
+
+    // When horizontal binning is used, horizontal decimation (if supported) is not
+    // available. When vertical binning is used, vertical decimation (if supported) is not
+    // available.
+
+    // If the camera is acquiring images, AcquisitionStop must be called before adjusting
+    // binning settings.
+    if (!binningConfiguration()) {
+        return false;
+    }
+
+    if (!decimationConfiguration()) {
+        return false;
+    }
+
+    if (!dimensionsConfiguration()) {
+        return false;
+    }
+
+    if (!exposureConfiguration()) {
+        return false;
+    }
+
+    Frame* frame = new Frame(_image_config.get().width,
+        _image_config.get().height,
+        _image_config.get().depth,
+        _image_config.get().format,
+        0,
+        0);
+    m_frame.reset(frame);
+
+    return true;
+}
+
+void Task::startCamera()
+{
+    LOG_INFO_S << "Starting frame acquisition." << endl;
+    m_device->StartStream();
+}
+
+void Task::acquireFrame()
+{
+    try {
+        Arena::IImage* image =
+            m_device->GetImage(_image_config.get().frame_timeout.toMilliseconds());
+        if (image->IsIncomplete()) {
+            LOG_ERROR_S << "Image is not complete!! " << endl;
+            m_device->RequeueBuffer(image);
+        }
+
+        size_t size = image->GetSizeFilled();
+        size_t width = image->GetWidth();
+        size_t height = image->GetHeight();
+        PfncFormat pixel_format = static_cast<PfncFormat>(image->GetPixelFormat());
+        uint8_t data_depth = 0U;
+        auto format = convertPixelFormatToFrameMode(pixel_format, data_depth);
+
+        Frame* out_frame = m_frame.write_access();
+        if (width != out_frame->getWidth() || height != out_frame->getHeight() ||
+            format != out_frame->getFrameMode()) {
+            out_frame->init(width, height, data_depth, format, 0, size);
+        }
+        out_frame->time = base::Time::now();
+        out_frame->received_time = out_frame->time;
+
+        out_frame->setImage(image->GetData(), size);
+        out_frame->setStatus(STATUS_VALID);
+        m_frame.reset(out_frame);
+        m_device->RequeueBuffer(image);
+        _frame.write(m_frame);
+    }
+    catch (GenICam::GenericException& ge) {
+        LOG_ERROR_S << "GenICam exception thrown: " << ge.what() << endl;
+    }
+    catch (std::exception& ex) {
+        LOG_ERROR_S << "Standard exception thrown: " << ex.what() << endl;
+    }
```

Comment:
We can't have a global catch over every single function.

It is normal for a Rock component to go into an EXCEPTION state on exception, so you should not be catching the exception *at all*.

Now, I do understand that one issue is that GenericException is not a subclass of std::exception. You should therefore have a try/catch only over the bits that call arena SDK methods and re-throw as a std::runtime_error with the same message.

Reply:
Comment valid for every single case where you did that. In the configureHook, you should be doing it as well, and get the call to TaskBase::configureHook out of the try.